src/software/ai/hl/stp/tactic/dribble/dribble_fsm.cpp

```diff
@@ -90,6 +90,36 @@ void DribbleFSM::getPossession(const Update &event)
                               event.common.world.field()) +
         Vector::createFromAngle(face_ball_orientation).normalize(0.05);
 
+    // if close to ball, pivot smartly
+    if (distance(event.common.robot.position(), ball_position) <
+        ROBOT_MAX_RADIUS_METERS * 5)
+    {
+        // find the target destination and orientation after the dribble
+        auto [target_destination, target_orientation] =
+            calculateNextDribbleDestinationAndOrientation(
+                event.common.world.ball(), event.common.robot,
+                event.control_params.dribble_destination,
+                event.control_params.final_dribble_orientation);
+
+        // if the target orientation is not close to the face ball orientation
+        // then we pivot around the ball
+        if ((target_orientation - event.common.robot.orientation()).clamp().abs() >
+            Angle::fromDegrees(20))
+        {
+            Angle target_angle;
+            Angle ang_1  = event.common.robot.orientation() + Angle::fromDegrees(30);
+            Angle ang_2  = event.common.robot.orientation() - Angle::fromDegrees(30);
+            target_angle = (ang_1 - target_orientation).clamp().abs() <
+                                   (ang_2 - target_orientation).clamp().abs()
+                               ? ang_1
+                               : ang_2;
```

Comment:
ang_1 and ang_2 should be renamed to be a more descriptive

also avoid using inline assignment, we prefer to use if statements here